multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+            .earning_wallet_info(EarningWalletInfo::Address(format!(
+                "{}",
+                node_wallet.clone()
+            )));
+        if let Some(port) = node.ui_port() {
+            config_builder = config_builder.ui_port(port)
+        }
+        if let Some(gas_price) = node.gas_price_opt() {
+            config_builder = config_builder.gas_price(gas_price)
+        }
+        eprintln!("{} wallet established: {}\n", node.name(), node_wallet,);
+        (config_builder.build(), node_wallet)
+    }
+
+    fn prepare_consuming_node(
+        &self,
+        cluster: &mut MASQNodeCluster,
+        blockchain_interfaces: &BlockchainInterfaces,
+    ) -> ConsumingNode {
+        let consuming_node_profile = self.test_inputs.node_profiles.consuming_node.clone();
+        let initial_service_fee_balance_minor =
+            consuming_node_profile.initial_service_fee_balance_minor;
+        let initial_tx_fee_balance_opt = consuming_node_profile.initial_tx_fee_balance_minor_opt;
+
+        let (consuming_node_config, consuming_node_wallet) =
+            self.get_node_config_and_wallet(&consuming_node_profile);
+        let initial_transaction_fee_balance = initial_tx_fee_balance_opt.unwrap_or(ONE_ETH_IN_WEI);
+        transfer_transaction_fee_amount_to_address(
+            &self.blockchain_params.contract_owner_wallet,
+            &consuming_node_wallet,
+            initial_transaction_fee_balance,
+            1,
+            &blockchain_interfaces.web3,
+        );
+        transfer_service_fee_amount_to_address(
+            self.blockchain_params.contract_owner_addr,
+            &self.blockchain_params.contract_owner_wallet,
+            &consuming_node_wallet,
+            initial_service_fee_balance_minor,
+            2,
+            &blockchain_interfaces.web3,
+            self.blockchain_params.chain,
+        );
+
+        assert_balances(
+            &consuming_node_wallet,
+            blockchain_interfaces.standard_blockchain_interface.as_ref(),
+            initial_transaction_fee_balance,
+            initial_service_fee_balance_minor,
+        );
+
+        let prepared_node = cluster.prepare_real_node(&consuming_node_config);
+        let consuming_node_connection = DbInitializerReal::default()
+            .initialize(&prepared_node.db_path, make_db_init_config(cluster.chain()))
+            .unwrap();
+        let consuming_node_payable_dao = PayableDaoReal::new(consuming_node_connection);
+        open_all_file_permissions(&prepared_node.db_path);
+        ConsumingNode::new(
+            consuming_node_profile,
+            prepared_node,
+            consuming_node_config,
+            consuming_node_wallet,
+            consuming_node_payable_dao,
+        )
+    }
+
+    fn prepare_serving_nodes(&self, cluster: &mut MASQNodeCluster) -> [ServingNode; 3] {
+        self.test_inputs
+            .node_profiles
+            .serving_nodes
+            .clone()
+            .into_iter()
```

Comment:
It's something that you pointed out, that we can get rid of the clone, since the argument used in the closure is only used in the form of the reference. So, you can simply use, `iter()` instead of `.clone().into_iter()`.